addons/delivery_shipper/models/delivery_carrier.py

```diff
@@ -0,0 +1,245 @@
+        shipper = self._get_shipper()
+        partner = order.partner_id if order and order.partner_id else origin
+
+        res = shipper._get_area_id(partner)
+        for area in res.get('data', []):
+            adm_level_5 = area.get('adm_level_5', {})
+
+            if (adm_level_5.get('postcode') == partner.zip):
+                return adm_level_5.get('id')
+        return False
+
+    def _shipper_get_package_information(self, order=False, picking=False):
+        original_weight_uom = self.env['product.template'].sudo()._get_weight_uom_id_from_ir_config_parameter()
+        target_weight_uom = self.env.ref('uom.product_uom_kgm')
+
+        total_weight, total_volume = 0, 0
+        if order:
+            for line in order.order_line.filtered(lambda line: not line.is_delivery and not line.display_type):
+                total_weight += line.product_uom_qty * line.product_id.weight
+                total_volume += line.product_uom_qty * line.product_id.volume
+        elif picking:
+            for move in picking.move_ids_without_package.filtered(lambda move: move.state not in ('done', 'cancel')):
+                total_weight += move.quantity * move.product_id.weight
+                total_volume += move.quantity * move.product_id.volume
+
+        estimated_dimension = math.pow(total_volume, (1/3)) if total_volume > 0 else 0
+
+        return {
+            'total_weight': original_weight_uom._compute_quantity(total_weight, target_weight_uom),
+            'estimated_dimension': estimated_dimension * 100,
+        }
+
+    def shipper_send_shipping(self, pickings, is_return=False):
+        shipper = self._get_shipper()
+        res = []
+        for picking in pickings:
+            if not picking.shipper_order_id:
+                # TODO: for "Send to Shipper" button after the delivery is validated and in state 'done' but cancelled
+                # need to add a check to trigger the create order here. The user should be allowed to do this
+                # can override the send_to_shipper to call create order first
+                raise UserError(_("You can only trigger a pickup after an order to Shipper is created. Please make sure to create it first."))
+
+            order_id = picking.shipper_order_id
+            if order_id:
+                order_result = {
+                    'exact_price': picking.shipper_order_amount,
+                    'tracking_number': order_id
+                }
+            res.append(order_result)
+            picking.carrier_tracking_ref = order_id
+
+            pickup_response = shipper._request_pickup(order_id)
+            logger.info(pickup_response)
+            picking.message_post(
+                body=_('Pickup request for %s was sent to Shipper with Order ID %s', picking.name, order_id)
+            )
+            label_response = shipper._get_shipping_label(order_id)
+            label_url = label_response.get('data', {}).get('url')
```

Comment:
Can't you do a GET request on the URL and fetch the label for the user directly?